pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -117,6 +124,33 @@ func New(port uint, runtime string) (Interface, error) {
 	return cadvisorClient, nil
 }
 
+func generateCollectorHttpClient(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile string) (*http.Client, error) {
+	//Enable accessing insecure endpoints. We should be able to access metrics from any endpoint
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+
+	if cAdvisorCollectorClientCertFile != "" {
+		if cAdvisorCollectorClientPrivateKeyFile == "" {
+			return nil, fmt.Errorf("invalid configuration: cAdvisorCollectorClientCertFile was specified and cAdvisorCollectorClientPrivateKeyFile was not specified.")
+		}
+		cert, err := tls.LoadX509KeyPair(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile)
+		if err != nil {
+			return nil, fmt.Errorf("Error loading the cAdvisor client certificates: %v", err)
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+		tlsConfig.BuildNameToCertificate()
```

Comment:
Yes, in particular its for accessing metric endpoints which have been configured to only accept client certificates signed with a particular CA which has a specific hostname.